htseq2multx.pl

```diff
@@ -0,0 +1,884 @@
+        exit(1);
+      }
+    elsif(fqmxVersionInsufficient($multx_version,$supported_version))
+      {
+        help2();
+        if(scalar(@$multx_version) < 1)
+          {print STDERR ('Unable to determine fastq-multx version of ',
+                         "executable [$fastq_multx].  Unable to proceed.\n")}
+        else
+          {print STDERR ('Version [',join('.',@$multx_version),
+                         "] of the fastq-multx executable [$fastq_multx] is ",
+                         "unsupported.  Only version [$FQMXVERSION] or higher ",
+                         'is supported.  Use the --fastq-multx option ',
+                         "described above to supply fastq-multx.\n")}
+        exit(1);
+      }
+
+    #Initialize the data structures/hashes used, e.g. index_hash, bchash,
+    #samplehash, and read in & convert eh barcode file
+    my $order = 1;
+    foreach my $fqf_num (@$idxread)
+      {$index_hash->{$fqf_num - 1} = $order++}
+    ($bcfilefqmx,$bchash,$samplehash) =
+      bcs2fqmxBCFile($bcfile,scalar(@$idxread));
+
+    ##
+    ## Process the prefix & suffix values
+    ##
+
+    if(!defined($prefix))
+      {$prefix = ''}
+
+    if(defined($prefix) && $galaxy)
+      {$prefix =~ s/_/-/g}
+
+    if(defined($suffix) && $galaxy)
+      {$suffix =~ s/_/-/g}
+
+    if(!defined($gzipin))
+      {$gzipin = ($fastq_files->[0] =~ /\.gz/)}
+
+    if(!defined($gzipout))
+      {
+        if(defined($suffix))
+          {$gzipout = ($suffix =~ /\.gz$/)}
+        else
+          {$gzipout = $gzipin}
+      }
+    elsif($gzipout)
+      {
+        if(defined($suffix) && $suffix !~ /\.gz$/)
+          {$suffix .= '.gz'}
+        elsif(!defined($suffix))
+          {$suffix = '.fastq.gz'}
+      }
+
+    if(!defined($suffix))
+      {$suffix = ($gzipout ? '.gz' : '.fastq')}
+
+    ##
+    ## Process the fastq file inputs
+    ##
+
+    #fastq-multx doesn't deal with gzipped input that doesn't have .gz in the
+    #name, so those files must be written to an actual file
+    if($gzipin)
+      {$tmp_files = processGZInfiles($fastq_files,$prefix)}
+
+    if($check_deflines)
+      {
+        $deflinesep = getDeflineSep($fastq_files);
+        if($deflinesep eq '')
+          {
+            if(scalar(@$fastq_files) > 1)
+              {print STDERR ('WARNING: Cannot identify fastq record type.  ',
+                             "Unable to check that deflines match.\n")}
+            $check_deflines = 0;
+          }
+      }
+
+    my $missing_infiles = [grep {!-e $_} @$fastq_files];
+    if(scalar(@$missing_infiles))
+      {
+        print STDERR ('ERROR: ',scalar(@$missing_infiles),' input files do ',
+                      'not exist: ',join(',',@$missing_infiles),"\n");
+        exit(1);
+      }
+
+    ##
+    ## Check the output files
+    ##
+
+    if(!$split_all && scalar(@$fastq_files) == scalar(@$idxread))
+      {$split_all = 1}
+
+    if(outfilesExist($bchash,$fastq_files,$split_all))
+      {exit(7)}
```

Comment:
Consider making some Readonly vars for various exit codes instead of hardcoding the exit code here. *e.g.*
```perl
Readonly::Scalar    $ERROR_OUTFILES_EXIST => 7;
```

```suggestion
      {exit($ERROR_OUTFILES_EXIST)}
```

Reply:
I was rather lax in emulating the exit codes in general, but in the cases where I do keep barcode splitter's exit codes, I think this is a good idea.

- [ ] Initialize globals for specific codes.